### PR TITLE
Compat Julia 0.7

### PR DESCRIPTION
**Pull request by wookay**
Compat 0.59.0 for `undef`

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -9,10 +9,7 @@ using WAV
...
 
-try
-    @testset ExtendedTestSet "SampledSignals Tests" begin
-        @includetests ARGS
-    end
-catch err
-    exit(-1)
+@testset ExtendedTestSet "SampledSignals Tests" begin
+    include.(["DummySampleStream.jl", "Interval.jl", "SampleBuf.jl", "SampleStream.jl", "SinSource.jl", "WAVDisplay.jl"])
+    # @includetests ARGS
```
[the remark this answers is not in the document]
Comment (wookay) [reply test/runtests.jl]: `@includetests ARGS` in Julia 0.7 does not run any tests on my computer. `TestSetExtensions` package needs  an update.
```
Test Summary:        |
SampledSignals Tests | No tests
```

**Comment by wookay** [comment]
thanks for the reivew. CI has failed on `SIUnits.jl` for Julia 0.7.

**Comment by ssfrr** [comment]
OK, thanks for the updates! I might need to do some work to merge this and @haberdashPI's PR #35 together. His PR migrates from SIUnits.jl to the newer and more actively-maintained [Unitful.jl](https://github.com/ajkeller34/Unitful.jl). I should be able to get to that sometime this week.

I'll also try to fix TestSetExtensions for 0.7 - I use it in most of my packages so it'll be nice to get it working.

**Comment by ssfrr** [comment]
Just an update - now that the Unitful PR is merged I've rebased this on top of it and I'm working out some of the issues. Hopefully I should get it merged soon.

**Comment by ssfrr** [comment]
Alrighty, we seem to be good on 0.7. I switched over to using LibSndFile.jl rather than WAV.jl in the testing and there's a little messiness to get testing working because of the circular dependency (for now we're using a branch of LibSndFile that has 0.7 fixes).

**Comment by haberdashPI** [comment]
Great! I'll start rebasing my mostly-finished `SampleFormat` proposal.

**Pull request merged**